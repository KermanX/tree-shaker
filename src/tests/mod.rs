use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
