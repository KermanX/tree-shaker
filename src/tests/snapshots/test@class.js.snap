---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/class.js
---
class A extends 2 {
	constructor(a, b) {
		super(a);
		b();
	}
	fn() {
		1();
	}
	[2] = 2;
	get [6]() {
		return 6;
	}
	set x(v) {
		this._x = v;
	}
}
new A();
export const a = class {
	[2] = 2;
};
export const b = class extends 2 {
	[2] = 2;
};
export default class {
	a = 1;
}
class __unused_848A {
	static {
		effect(2);
		effect(4);
		effect(7);
		effect(1);
		effect(5);
		{
			effect(6);
		}
	}
}
